devito/tools/data_structures.py

```diff
@@ -29,18 +30,33 @@ def __init__(self, **kwargs):
         self.__dict__.update(kwargs)
 
 
-class EnrichedTuple(tuple):
+class EnrichedTuple(Pickable, tuple):
```

Comment:
this should be `Reconstructable` not `Pickable`

but even then

why is it needed at all? Can you write an MFE?

Reply:
The MFE is the new test. And consequently all halo/padding are reconstructed wrong